forum/communications/mongo_models.py

```diff
@@ -27,7 +27,34 @@ class Notification(BaseTimestampModel):
     initiator = fields.EmbeddedDocumentField(NamespaceInfo, required=True)
     receivers = fields.EmbeddedDocumentListField(NamespaceInfo, required=True)
     message = fields.StringField(required=True, max_length=255)
-    
+
+
+class NotificationTypes(Document):
+    name = fields.StringField(required=True, unique=True, max_length=50)
+    description = fields.StringField(max_length=255)
+
+
+class NotificationPreferences(Document):
+    user_id = fields.LongField(required=True)
+    notification_types = fields.EmbeddedDocumentListField(NotificationTypes)
+
+    ws_enabled = fields.BooleanField(default=True)
+    email_enabled = fields.BooleanField(default=True)
+
+    @classmethod
+    def is_ws_enabled(cls, user_id, notification_type_name):
+        preference = cls.objects(user_id=user_id, notification_types_name=notification_type_name).first()
+        return preference.ws_enabled if preference else False
+
+    @classmethod
+    def is_email_enabled(cls, user_id, notification_type_name):
+        preference = cls.objects(user_id=user_id, notification_types_name=notification_type_name).first()
+        return preference.email_enabled if preference else False
```

Comment:
The query methods `is_ws_enabled` and `is_email_enabled` might be inefficient as they perform a search with the field `notification_types_name`, which doesn't match any field in the model.
Update queries to match existing fields or change the method to better reflect the schema.